lib/course_planner_web/controllers/offered_course_controller.ex

```diff
@@ -86,13 +86,54 @@ defmodule CoursePlannerWeb.OfferedCourseController do
                               user_role: user_role)
   end
 
+  def edit(%{assigns: %{current_user: %{role: user_role} = current_user}} = conn, %{"id" => id})
+  when user_role == "Teacher" do
+    {:ok, offered_course, changeset} = OfferedCourses.load_offered_course_for_edit(id)
+
+    if OfferedCourses.can_teacher_edit_and_update_offered_course?(current_user, offered_course) do
+      render(conn, "teacher_edit.html", offered_course: offered_course, changeset: changeset)
+    else
+      conn
+      |> put_status(403)
+      |> render(CoursePlannerWeb.ErrorView, "403.html")
+    end
+  end
+
   def edit(conn, %{"id" => id}) do
+    {:ok, offered_course, changeset} = OfferedCourses.load_offered_course_for_edit(id)
+
+    render(conn, "edit.html", offered_course: offered_course, changeset: changeset)
+  end
+
+  def update(%{assigns: %{current_user: %{role: user_role, id: user_id}}} = conn,
+    %{"id" => id, "offered_course" => %{"syllabus" => syllabus}}) when user_role == "Teacher" do
+
     offered_course =
       OfferedCourse
       |> Repo.get!(id)
       |> Repo.preload([:term, :course, :students, :teachers])
-    changeset = OfferedCourse.changeset(offered_course)
-    render(conn, "edit.html", offered_course: offered_course, changeset: changeset)
+
+      can_user_update_this_offered_course? =
+        offered_course.teachers
+        |> Enum.any?(fn(teacher) -> teacher.id ==  user_id end)
+
+      if can_user_update_this_offered_course? do
```

Comment:
you should use the function that you extracted here `OfferedCourses.can_teacher_edit_and_update_offered_course?(current_user, offered_course)`

Reply:
done